src/movies.js

```diff
@@ -1,29 +1,114 @@
 // The `movies` array from the file `src/data.js`.
-console.log('movies: ', movies);
+// console.log('movies: ', movies);
+
+const movies = require("./data");
 
 
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(moviesArr) {
+
+  const directorsArr = moviesArr.map(directorName=> directorName.director)
+  return directorsArr
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(moviesArr) {
+
+  const spielbergDramaMovies = moviesArr.filter(movie =>{
+    return (movie.director === 'Steven Spielberg' && movie.genre.indexOf('Drama') !== -1 ) 
+  })
+  return spielbergDramaMovies.length
+}
+
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(moviesArr) {
+
+  if(!moviesArr.length) return 0 ;
+  else{
+    const sumScores = moviesArr.reduce((a,b)=>{
+      if(!b.score){
+        b.score = 0;
+        return a + b.score ;
+      }else{
+        return a + b.score ;
+      }
```

Comment:
if `b.score` is undefined/does not exist - it is enough to` return a`, will be shorter syntax.

```
      if(!b.score){
        return a ;
      }else{
        return a + b.score ;
      }
```

A challenge for you as a **bonus** -> read about **ternary operator,** and try to convert the if statement to new syntax . If not clear let me know!
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator

`condition ? exprIfTrue : exprIfFalse`